internal/config/types.go

```diff
@@ -8,14 +8,15 @@ package config
 import "time"
 
 type Config struct {
-	Version            string         `yaml:"-"`
-	Path               string         `yaml:"-"`
-	Log                Log            `yaml:"-" mapstructure:"log"`
-	ProcessMonitor     ProcessMonitor `yaml:"-" mapstructure:"process_monitor"`
-	DataplaneAPI       DataplaneAPI   `yaml:"-" mapstructure:"dataplane_api"`
-	Client             Client         `yaml:"-" mapstructure:"client"`
-	ConfigDir          string         `yaml:"-" mapstructure:"config-dirs"`
-	AllowedDirectories []string       `yaml:"-"`
+	Version            string          `yaml:"-"`
+	Path               string          `yaml:"-"`
+	Log                Log             `yaml:"-" mapstructure:"log"`
+	ProcessMonitor     ProcessMonitor  `yaml:"-" mapstructure:"process_monitor"`
+	DataplaneAPI       DataplaneAPI    `yaml:"-" mapstructure:"dataplane_api"`
+	DataPlaneConfig    DataPlaneConfig `yaml:"-" mapstructure:"dataplane_config"`
```

Comment:
```suggestion
	DataPlaneAPI       DataPlaneAPI    `yaml:"-" mapstructure:"data_plane_api"`
	DataPlaneConfig    DataPlaneConfig `yaml:"-" mapstructure:"data_plane_config"`
```

Also need to update the `DataplaneAPI` struct to `DataPlaneAPI`.

Reply:
I didn't actually add `DataplaneAPI` in this PR. I have a separate task to rename everything from `Dataplane` to `DataPlane` so I'll make that update in a separate PR.

Reply:
Have separate PR here for this https://github.com/nginx/agent/pull/579